ext/http/01_http.js

```diff
@@ -290,13 +292,40 @@
               }
               await reader.cancel(error);
               throw error;
+            } finally {
+              readableStreamClose(respBody); // Release JS lock.
```

Comment:
Does this throw if called after `await reader.cancel()`?

Reply:
It's unreachable since `catch` throws after `await reader.cancel()`.

Reply:
No, `finally` will still run after that `throw`: `try { throw "a" } catch { throw "b" } finally { console.log("c") }`.

Reply:
ooh TIL. `readableStreamClose` does throw, since `reader.cancel` closes it already. Maybe I should move it back to the `try` block?

Reply:
SGTM